src/Security/RootPermissions.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace SilverStripe\Security;
+
+/**
+ * For root objects that have permission type "Inherit" determine the result of certain permission checks
```

Comment:
Also, I'm wondering if we should just name this interface `PermissionChecked` and apply it to `DataObject` as well? There's nothing in the interface which makes this "root only", right? Which also means that you assign the object class itself here, which means you don't want to distinguish between "root level" and "instance level" permission checks in your implementation.